project/Scope.rst

```diff
@@ -0,0 +1,23 @@
+Scope of Halium
+===============
+
+Halium's scope primarily consists of 2 parts:
+
+1. Halium's Android is a minimal Android distribution designed to run within an LXC container. It will provide a set of interfaces to let the host of LXC container utilize Android's HAL to communicate with the hardware of Android-based smartphones. These interfaces come in 3 forms:
+
+    1. Native Android interfaces. (e.g. rild's socket, Audio HAL (dlopen'ed directly), (original) WiFi HAL (a standard libhybris wrapper))
+    2. Libhybris compat layer. Usually used to wrap Android C++ interfaces, contains some logic, and has its counter-part compiled in the Android tree. (e.g. HWC2 compat layer)
+    3. Custom interfaces. This will contain more than a compat layer and may have quite deep integration within Android code. (e.g. Ubuntu Touch's camera interfaces, which consists of camera compat layer, custom media compat layer, and a deep modification of Android's libmedia). If possible, we would like to reduce this kind of interface as much as possible to reduce our workload, especially those doing deep modification to Android.
+
+2. Halium's middleware is a set of software/libraries/modules on GNU/Linux which talks to the interfaces in 1. and is distro- & toolkit-agnostic enough. For example, this includes pulseaudio-module-droid, but will not include qtubuntu-camera (a client of the aforementioned camera interface, because it's tied to Qt). Halium might not be upstream of most of the middlewares. Thus, most of the time we'll just maintain the minimal patch(es) that make them work in Halium-based environment.
```

Comment:
Replace "Distro and toolkit-agnostic enough" with "Mostly distro and toolkit agnostic". Also expand & to "and" for clarity.